app/views/topics/show.html.erb

```diff
@@ -3,4 +3,6 @@
 
   <h2> <%= @topic.name %></h2>
 
+  <%= button_to 'Delete Topic', {action: "destroy", id:(@topic.id)},method: :delete, data: { confirm: "Are you sure?" } %>
```

Comment:
You don't need the brackets round the `@topic.id`.
Another approach is to use `topic_path(@topic)`, instead of `{action: "destroy, id:(@opic.id)}`